File: DSM2/TPI/exercicios/Fornecedor/src/main/java/com/mycompany/fornecedor/Fornecedor.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.fornecedor;

import java.util.Scanner;

/**
 *
 * @author fatec-dsm2
 */
public class Fornecedor {

    private String nomeFornecedor;
    private String nomeProduto;
    private String descricaoProduto;

    public String getNomeFornecedor() {
        return nomeFornecedor;
    }

    public void setNomeFornecedor(String nomeFornecedor) {
        this.nomeFornecedor = nomeFornecedor;
    }

    public String getNomeProduto() {
        return nomeProduto;
    }

    public void setNomeProduto(String nomeProduto) {
        this.nomeProduto = nomeProduto;
    }

    public String getDescricaoProduto() {
        return descricaoProduto;
    }

    public void setDescricaoProduto(String descricaoProduto) {
        this.descricaoProduto = descricaoProduto;
    }

    public void cadastrarFornecedor(String nomeProduto) {

        Scanner scanner = new Scanner(System.in);

        setNomeProduto(nomeProduto);

        System.out.println("Insira o nome do fornecedor");
        setNomeFornecedor(scanner.nextLine());

        System.out.println("Insira a decricao do produto");
        setDescricaoProduto(scanner.nextLine());

    }

    public String listarFornecedor() {

        return "Fornecedor: " + getNomeFornecedor()
                + "\nProduto: " + getNomeProduto()
                + "\nDescricao: " + getDescricaoProduto();

    }

}

File: DSM3/TP2/aula04_OO/exercicios/Fornecedor/Fornecedor.py
class Fornecedor:
    def __init__(self):
        self.__nomeFornecedor = ""
        self.__nomeProduto = ""
        self.__descricao = ""

    @property
    def _nomeFornecedor(self):
        return self.__nomeFornecedor

    @_nomeFornecedor.setter
    def _nomeFornecedor(self, value):
        self.__nomeFornecedor = value

    @property
    def _nomeProduto(self):
        return self.__nomeProduto

    @_nomeProduto.setter
    def _nomeProduto(self, value):
        self.__nomeProduto = value

    @property
    def _descricao(self):
        return self.__descricao

    @_descricao.setter
    def _descricao(self, value):
        self.__descricao = value

    def cadastrarFornecedor(self):
        print("\nCadastro de fornecedor")
        self._nomeFornecedor = input("Insira o nome do fornecedor: ")
        self.__nomeProduto = input("Insira o nome do produto: ")
        self._descricao = input("Insira a descrição: ")

    def listarFornecedor(self):
        print("==Dados do Fornecedor==")
        print(f"""
        Fornecedor: {self._nomeFornecedor}
        Produto: {self._nomeProduto}
        Descrição: {self._descricao}
        """)
